samples: benchmarks: coremark: add FLPR support for nRF54H20 DK (#20614)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

  * Added:

    * Support for the nRF54L05 and nRF54L10 SoCs (emulated on nRF54L15 DK).

>     * FLPR core support for the :ref:`zephyr:nrf54l15dk_nrf54l15` and the :ref:`zephyr:nrf54h20dk_nrf54h20` board target.

Reviewer: ```suggestion
    * FLPR core support for the :ref:`zephyr:nrf54l15dk_nrf54l15` and :ref:`zephyr:nrf54h20dk_nrf54h20` board targets.
```

Author: fixed

  * Added:

    * Support for the nRF54L05 and nRF54L10 SoCs (emulated on nRF54L15 DK).
    * FLPR core support for the :ref:`zephyr:nrf54l15dk_nrf54l15` and :ref:`zephyr:nrf54h20dk_nrf54h20` board targets.

  * Removed the following compiler options that were set in the :kconfig:option:`CONFIG_COMPILER_OPT` Kconfig option:

    * ``-fno-pie``
    * ``-fno-pic``
    * ``-ffunction-sections``
    * ``-fdata-sections``

>     These options are enabled by default in Zephyr and do not need to be set with the dedicated Kconfig option.

Reviewer 2: ```suggestion
    These options are enabled by default in Zephyr.
    You do not need to set them with a dedicated Kconfig option.
```

Author: discussed in the other comm channel, decide to keep it as it is
